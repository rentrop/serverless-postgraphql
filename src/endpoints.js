
var endpoint = 'http://localhost:5000/graphql';

if(process.env.USE_AWS_ENDPOINT) {
  endpoint = process.env.AWS_ENDPOINT;
};

module.exports = {
    endpoint: endpoint
};
