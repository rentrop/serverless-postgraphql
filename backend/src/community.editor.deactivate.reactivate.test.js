import HttpTransport from 'lokka-transport-http';
import Lokka from 'lokka';
import uuidv4 from 'uuid';
import { endpoint } from './endpoints';

const anonLokka = new Lokka({transport: new HttpTransport(endpoint)});
const superAdminEmail = 'superadmin@flo.ods';
const communityEditorEmail = 'editor@community.floods';
const superAdminPassword = 'texasfloods';
const newUserPassword = 'texasfloods';

async function getToken(email, password) {
  const response = await anonLokka.send(`
    mutation($email:String!, $password:String!) {
      authenticate(input: {email: $email, password: $password}) {
        jwtToken
      }
    }
  `,
  {
    email: email,
    password: password
  });

  return response.authenticate.jwtToken;
}

describe('When deactivating, and reactivating a user as a community editor', () => {
  var newUserEmail = uuidv4() + '@flo.ods';
  var newUserId;

  describe('As a super admin', async () => {
    var lokka;

    beforeAll(async (done) => {
      getToken(superAdminEmail, superAdminPassword).then((token) => {
        const headers = {
          'Authorization': 'Bearer '+ token
        };
        lokka = new Lokka({transport: new HttpTransport(endpoint, {headers})});
        done();
      });
    });

    it('should register a new community editor', async () => {
      const response = await lokka.send(`
        mutation($email:String!) {
          registerUser(input: {
            firstName: "New",
            lastName: "User",
            jobTitle: "Community Editor",
            communityId: 1,
            phoneNumber: "555-6666",
            email: $email,
            password:"texasfloods",
            role:"floods_community_editor"
          }) {
            user {
              id
            }
          }
        }
      `,
      {
        email: newUserEmail
      });

      newUserId = response.registerUser.user.id;
      expect(response).not.toBeNull();
    });
  });

  describe('As a community editor', async () => {
    var lokka;

    beforeAll(async (done) => {
      getToken(communityEditorEmail, superAdminPassword).then((token) => {
        const headers = {
          'Authorization': 'Bearer '+ token
        };
        lokka = new Lokka({transport: new HttpTransport(endpoint, {headers})});
        done();
      });
    });

    it('should fail to deactivate the new community editor', async () => {
      try {
        const response = await lokka.send(`
          mutation($userID:Int!) {
            deactivateUser(input: {userId: $userID}) {
              user {
                id
              }
            }
          }
        `,
        {
          userID: newUserId
        });
      } catch(e) {
        expect(e).toMatchSnapshot();
      }
    });
  });

  describe('As a super admin again', async () => {
    var lokka;

    beforeAll(async (done) => {
      getToken(superAdminEmail, superAdminPassword).then((token) => {
        const headers = {
          'Authorization': 'Bearer '+ token
        };
        lokka = new Lokka({transport: new HttpTransport(endpoint, {headers})});
        done();
      });
    });

    it('should deactivate the user', async () => {
      const response = await lokka.send(`
        mutation($userID:Int!) {
          deactivateUser(input: {userId: $userID}) {
            user {
              id
            }
          }
        }
      `,
      {
        userID: newUserId
      });

      expect(response).not.toBeNull();
    });
  });

  describe('As a community editor again', async () => {
    var lokka;

    beforeAll(async (done) => {
      getToken(communityEditorEmail, superAdminPassword).then((token) => {
        const headers = {
          'Authorization': 'Bearer '+ token
        };
        lokka = new Lokka({transport: new HttpTransport(endpoint, {headers})});
        done();
      });
    });

    it('should fail to reactivate the new community editor', async () => {
      try {
      const response = await lokka.send(`
          mutation($userId:Int!, $email:String!) {
            reactivateUser(input: {
              userId: $userId,
              email: $email
              password: "texasfloods"
              role: "floods_community_editor"
            }) {
              user {
                firstName
                lastName
                active
              }
            }
          }
        `,
        {
          userId: newUserId,
          email: newUserEmail
        });
      } catch(e) {
        expect(e).toMatchSnapshot();
      }
    });
  });
});
