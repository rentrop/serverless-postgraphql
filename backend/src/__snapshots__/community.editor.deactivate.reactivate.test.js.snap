// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When deactivating, and reactivating a user as a community editor As a community editor again should fail to reactivate the new community editor 1`] = `[Error: GraphQL Error: permission denied for function reactivate_user]`;

exports[`When deactivating, and reactivating a user as a community editor As a community editor should fail to deactivate the new community editor 1`] = `[Error: GraphQL Error: Community editors can only deactivate themselves]`;
